Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/ScheduledRides/ScheduledRideAssignedOnUpdate.hs

```diff
@@ -0,0 +1,214 @@
+import qualified Domain.Types.Person as DPerson
+import qualified Domain.Types.Ride as DRide
+import Kernel.Beam.Functions (runInReplica)
+import Kernel.External.Maps.Types (LatLong (..))
+import Kernel.Prelude
+import Kernel.Storage.Esqueleto.Config (EsqDBReplicaFlow)
+import qualified Kernel.Storage.Hedis as Redis
+import Kernel.Streaming.Kafka.Producer.Types (KafkaProducerTools)
+import Kernel.Types.Id
+import Kernel.Utils.Common
+import Lib.Scheduler
+import Lib.SessionizerMetrics.Types.Event
+import SharedLogic.Allocator
+import SharedLogic.CallBAP
+import qualified SharedLogic.External.LocationTrackingService.Flow as LTF
+import qualified SharedLogic.External.LocationTrackingService.Types as LT
+import SharedLogic.GoogleTranslate (TranslateFlow)
+import qualified Storage.Cac.TransporterConfig as SCTC
+import qualified Storage.Queries.Booking as QBooking
+import qualified Storage.Queries.DriverInformation as QDI
+import qualified Storage.Queries.DriverInformationExtra as QDE
+import qualified Storage.Queries.Person as QP
+import qualified Storage.Queries.Ride as QRide
+import qualified Storage.Queries.Vehicle as QVeh
+import qualified Tools.Maps as TMaps
+import qualified Tools.Metrics as Metrics
+import TransactionLogs.Types
+
+sendScheduledRideAssignedOnUpdate ::
+  ( MonadFlow m,
+    EsqDBFlow m r,
+    EncFlow m r,
+    HasHttpClientOptions r c,
+    HasShortDurationRetryCfg r c,
+    CacheFlow m r,
+    HasField "modelNamesHashMap" r (HMS.HashMap Text Text),
+    HasFlowEnv m r '["nwAddress" ::: BaseUrl],
+    HasField "s3Env" r (S3.S3Env m),
+    LT.HasLocationService m r,
+    HasFlowEnv m r '["ondcTokenHashMap" ::: HMS.HashMap KeyConfig TokenConfig],
+    HasFlowEnv m r '["internalEndPointHashMap" ::: HMS.HashMap BaseUrl BaseUrl],
+    HasFlowEnv m r '["kafkaProducerTools" ::: KafkaProducerTools],
+    EsqDBReplicaFlow m r,
+    HasField "searchRequestExpirationSeconds" r NominalDiffTime,
+    HasField "jobInfoMap" r (M.Map Text Bool),
+    HasField "maxShards" r Int,
+    HasField "schedulerSetName" r Text,
+    HasField "schedulerType" r SchedulerType,
+    Metrics.HasSendSearchRequestToDriverMetrics m r,
+    HasLongDurationRetryCfg r c,
+    HasField "singleBatchProcessingTempDelay" r NominalDiffTime,
+    TranslateFlow m r,
+    HasFlowEnv m r '["maxNotificationShards" ::: Int],
+    Redis.HedisFlow m r,
+    EventStreamFlow m r,
+    HasField "minTripDistanceForReferralCfg" r (Maybe HighPrecMeters),
+    Metrics.HasCoreMetrics r
+  ) =>
+  Job 'ScheduledRideAssignedOnUpdate ->
+  m ExecutionResult
+sendScheduledRideAssignedOnUpdate Job {id, jobInfo} = withLogTag ("JobId-" <> id.getId) do
+  let jobData = jobInfo.jobData
+      bookingId = jobData.bookingId
+      driverId = jobData.driverId
+      rideId = jobData.rideId
+  mbRide <- QRide.findById rideId
+  mbDriverInfo <- QDE.findById driverId
+  case (mbRide, mbDriverInfo) of
+    (Nothing, _) -> return $ Terminate "Job Terminated because Ride not found"
```

Comment:
Shouldn't we reallocate the booking in this case too?

Reply:
By the way, what if driver/customer cancels the ride before this job gets execute? Is it handled?

Reply:
this would be visible in next changes

Reply:
current flow also reallocate only if ride is present